src/MUIDataTable.js

```diff
@@ -413,7 +416,8 @@ class MUIDataTable extends React.Component {
       }
 
       if (column.filterOptions) {
-        filterData[colIndex] = cloneDeep(column.filterOptions);
+        if (Array.isArray(column.filterOptions)) filterData[colIndex] = cloneDeep(column.filterOptions);
```

Comment:
Sorry I missed it earlier, but we should add a console error to warn people about the deprecation of the array form since we will eventually be moving away from that.

```
if (Array.isArray(column.filterOptions)) {
  console.error(
    'Deprecated: filterOptions must now be an object. see https://github.com/gregnb/mui-datatables/tree/master/examples/customize-filter example',
  );
}
```